test/dynamo/test_activation_checkpointing.py

```diff
@@ -1000,6 +1002,98 @@ def fn(x, y):
         ):
             self._validate(fn, backend, x, y)
 
+    @torch._dynamo.config.patch(inline_inbuilt_nn_modules=True)
+    def test_compile_selective_checkpoint_reparameterization(self):
+        def sac_policy():
+            def _recomp_policy():
+                def _custom_policy(ctx, func, *args, **kwargs):
+                    to_recompute = func in {
+                        torch.ops.aten.mul.Tensor,
+                        torch.ops.aten.sigmoid.default,
+                    }
+                    return (
+                        CheckpointPolicy.MUST_RECOMPUTE
+                        if to_recompute
+                        else CheckpointPolicy.MUST_SAVE
+                    )
+
+                return _custom_policy
+
+            return create_selective_checkpoint_contexts(_recomp_policy())
+
+        class Reparam(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def reparam(self, x):
+                return torch.sigmoid(torch.mul(x, x))
+
+            def forward(self, x):
+                return checkpoint(
+                    self.reparam, x, use_reentrant=False, context_fn=sac_policy
+                )
+
+        def apply_reparam(model):
+            modules = list(model.modules())
+
+            for mod in modules:
+                params_dict = dict(mod.named_parameters(recurse=False))
+                for p_name, p in params_dict.items():
+                    mod.register_parameter(p_name, nn.Parameter(p))
+                    nn.utils.parametrize.register_parametrization(
+                        mod, p_name, Reparam(), unsafe=True
+                    )
+
+            return model
+
+        class MLPModule(nn.Module):
+            def __init__(self):
+                super().__init__()
+                torch.manual_seed(5)
+                self.net1 = nn.Linear(16, 16, bias=False)
+
+            def forward(self, x):
+                return self.net1(x)
+
+            def reset_parameters(self):
+                self.net1.reset_parameters()
+
+        fw_compiler = functools.partial(
+            count_ops,
+            freqs=[1, 1],
+            ops=[torch.ops.aten.mul.Tensor, torch.ops.aten.sigmoid.default],
```

Comment:
maybe noob question: where would the `sigmoid` be called?

Reply:
The `sigmoid` is called in `return torch.sigmoid(torch.mul(x, x))` above in the parametrization module

Reply:
my bad, didn't see it, thx!